src/node/node_class.hpp

```diff
@@ -37,256 +57,1269 @@ using IndexPropertyType = js::IndexPropertyType<Types>;
+		return;
+	}
+
+	try {
+		node::Types::FunctionCallback constructor_callback = (node::Types::FunctionCallback)info.Data();
+		constructor_callback(info);
+	}
+	catch (const node::Exception& e) {
+		Napi::Error error = Napi::Error::New(info.Env(), e.what());
+		copyObject(env, e.m_value, error);
+		throw error;
+	}
+	catch (const std::exception& e) {
+		throw Napi::Error::New(env, e.what());
+	}
+}
+
+template<typename ClassType>
+Napi::Function WrappedObject<ClassType>::init(Napi::Env env, 
+	const std::string& name, 
+	node::Types::FunctionCallback constructor_callback, 
+	std::function<bool(const std::string&)> has_native_method_callback,
+	const std::vector<Napi::ClassPropertyDescriptor<WrappedObject<ClassType>>>& properties, 
+	const IndexPropertyType* indexPropertyHandlers) {
+
+	m_name = name;
+	m_has_native_methodFunc = has_native_method_callback;
+	WrappedObject<ClassType>::m_nullExternal = Napi::Persistent(Napi::External<Internal>::New(env, nullptr));
+
+	Napi::Function ctor = Napi::ObjectWrap<WrappedObject<ClassType>>::DefineClass(env, name.c_str(), properties, (void*)constructor_callback);
+	
+	constructor = Napi::Persistent(ctor);
+	constructor.SuppressDestruct();
+
+	m_indexPropertyHandlers = const_cast<IndexPropertyType*>(indexPropertyHandlers);
+	return ctor;
+}
+
+//This creates the required JS instance with a Proxy parent to support get and set handlers and returns a proxy created on the JS instance to support property enumeration handler
+//the returned JS Proxy has only ownKeys trap setup so that all other member accesses skip the Proxy and go directly to the JS instance
+template<typename ClassType>
+Napi::Value WrappedObject<ClassType>::create_instance_with_proxy(const Napi::CallbackInfo& info) {
+	Napi::Env env = info.Env();
+
+	if (constructor.IsEmpty()) {
+		std::string typeName(typeid(ClassType).name());
+		std::string errorMessage = "create_instance_with_proxy: Class " + typeName + " not initialized. Call init() first";
+		throw Napi::Error::New(env, errorMessage);
+	}
+
+	if (!info.IsConstructCall()) {
+		throw Napi::Error::New(env, "This function should be called as a constructor");
+	}
+
+	Napi::EscapableHandleScope scope(env);
+
+	try {
+
+		auto arguments = napi_get_arguments(info);
+		Napi::Object instance = constructor.New(arguments);
+
+		//using DefineProperty to make it non enumerable and non configurable and non writable
+		instance.DefineProperty(Napi::PropertyDescriptor::Value("_instance", instance, napi_default));
+		
+		info.This().As<Napi::Object>().DefineProperty(Napi::PropertyDescriptor::Value("isRealmCtor", Napi::Boolean::New(env, true), napi_configurable)); 
+		
+		ObjectSetPrototypeOf.Call({ info.This(), instance });
+		Napi::Object instanceProxy = GlobalProxy.New({ info.This(), ProxyHandler::get_instance_proxy_handler(env) });
+		
+		instance.DefineProperty(Napi::PropertyDescriptor::Value("_instanceProxy", instanceProxy, napi_default));
+		return scope.Escape(instanceProxy);
+	}
```

Comment:
I think it would be better if this block is stored values using JS symbols, instead of the strings `"_instance"`, `"isRealmCtor"`, `"_instanceProxy"`.

Reply:
This was implemented before I have used a symbol for the external value and was not sure how performant this would be. I will see if I can refactor it to to use symbols and if there are any implications of that

Reply:
I suggest we keep this as underscore strings key members since having these with a Symbol will affect the performance

Reply:
I don't see how it could affect performance, if the string to symbol translation is done only once? But fair enough, I trust that you've done the measurements.